test/node/index.js

```diff
@@ -116,6 +116,49 @@ describe('parser', function() {
     });
   });
 
+  it('should decode an ArrayBuffer/Buffer with undeclared type as binary', function(done) {
+    var buffer1 = new ArrayBuffer(4);
+    var dataview = new DataView(buffer1);
+    var buffer2 = new Buffer(4);
+    for (var i = 0; i < buffer1.byteLength ; i++) {
+      dataview.setInt8(i, 4-i);
+      buffer2[i] = 4-i;
+    }
+
+    var decoded1 = decode(buffer1, null);
+    var decoded2 = decode(buffer2, null);
+    expect(decoded1).to.eql({ type: 'message', data: new Buffer([3,2,1])});
+    expect(new Uint8Array(decoded1.data)).to.eql(new Uint8Array(buffer1.slice(1)));
+    expect(decoded2).to.eql({ type: 'message', data: new Buffer(buffer2.slice(1))});
+    expect(new Uint8Array(decoded2.data)).to.eql(new Uint8Array(buffer2.slice(1)));
+    done();
+  });
+
+  it('should decode an ArrayBuffer/Buffer as binary of type ArrayBuffer', function(done) {
```

Comment:
It tests that Buffer and ArrayBuffer both work as in- and output of decode(), which is what this fix is about (the intent that it fixes cross-platform issues in Socket.io and Node.js/nw.js/Electron when they use encode()/decode() because they can be of both types)
Therefore, I think the test is okay as it is in creating a Buffer and ArrayBuffer, and testing it once with arrayBuffer as binaryType and another time with no binaryType (might as well be "message" though, afaik).

If this is beyond the intended scope of the function, please reject this PR.